### PR TITLE
685 - Include swap fee in the chart price calculations

### PR DESCRIPTION
**Pull request by giladt**

### CODE REVIEW

File: src/services/ProjectTokenHistoricalPriceService.ts [hunk 1]
```diff
@@ -237,17 +237,19 @@ export class ProjectTokenHistoricalPriceService {
     const poolPTBalance = this.numberService.fromString(fromWei(lbpMgr.lbp.vault.projectTokenBalance, lbpMgr.projectTokenInfo.decimals));
     const poolFTBalance = this.numberService.fromString(fromWei(lbpMgr.lbp.vault.fundingTokenBalance, lbpMgr.fundingTokenInfo.decimals));
     const projectTokenAmount = this.numberService.fromString(lastSwap.tokenAmountOut);
-    // const fundingTokenAmount = this.numberService.fromString(lastSwap.tokenAmountIn);
     const forecastData = await this.priceService.getInterpolatedPriceDataPoints(
       poolPTBalance,
-      poolFTBalance,
+      poolFTBalance * (1 + lbpMgr.swapFeePercentage),
```
Comment (fabianschu) [inline src/services/ProjectTokenHistoricalPriceService.ts:242]: I don't think this is necessary. When we get the `poolFTBalance` from the vault, my understanding is that all swap fees that had been collected so far are included in that `poolFTBalance`.

File: src/services/ProjectTokenHistoricalPriceService.ts [hunk 2]
```diff
@@ -159,31 +159,31 @@ export class ProjectTokenHistoricalPriceService {
...
+            (this.numberService.fromString(todaysSwaps[todaysSwaps.length-1].tokenAmountIn) * (1 + lbpMgr.swapFeePercentage)) /
             (this.numberService.fromString(todaysSwaps[todaysSwaps.length-1].tokenAmountOut)) *
-            priceAtTimePoint
+            usdPriceAtTimePoint
           ),
         });
 
         previousTimePoint = (
-          (this.numberService.fromString(todaysSwaps[todaysSwaps.length-1].tokenAmountIn)) /
+          (this.numberService.fromString(todaysSwaps[todaysSwaps.length-1].tokenAmountIn) * (1 + lbpMgr.swapFeePercentage)) /
```
Comment (fabianschu) [inline src/services/ProjectTokenHistoricalPriceService.ts:173]: Just to make sure: the swap fee is not already included in the swap data that we receive from the subgraph per swap?

Comment (dkent600) [reply src/services/ProjectTokenHistoricalPriceService.ts:173]: correct, we confirmed this looking at the contracts.  wouldn't hurt if you want to triple check @fabianschu .

File: src/services/ProjectTokenHistoricalPriceService.ts (hunk 1 above)
Comment (dkent600) [reply src/services/ProjectTokenHistoricalPriceService.ts:242]: You are correct @fabianschu

Comment (giladt) [reply src/services/ProjectTokenHistoricalPriceService.ts:242]: removed the fee from pool balance

**Review by dkent600 — APPROVED** [review]

**Review by fabianschu — COMMENTED** [review]
looks good to me

**Pull request merged**